typing/typecore.ml

```diff
@@ -3230,7 +3230,12 @@ and type_expect_
       re { exp with exp_extra =
              (Texp_poly cty, loc, sexp.pexp_attributes) :: exp.exp_extra }
   | Pexp_newtype({txt=name}, sbody) ->
-      let ty = newvar () in
+      let ty =
+        if name <> "" && name.[0] = '_' then (* Invalid variable name. *)
```

Comment:
`Typetexp` (or Types,....?) could expose a `val valid_tyvar_name: string -> bool` (and use it internally in transl_type_param).